### ISSUE TITLE
Intermittent failure on enable docker task

### ISSUE THREAD

**Issue by hpreston** [mantl/mantl#984]
I've had several instances where running the setup playbook errors the first time running, and then running it immediately again it completes fine, or at least makes more progress.  The step that most often causes the problem on the first run is at this point:  

```
TASK: [docker | enable docker] ************************************************ 
[0;31mfailed: [test-control-03] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31mfailed: [test-worker-001] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31mfailed: [test-control-02] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31mfailed: [test-edge-01] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31mfailed: [test-edge-02] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31mfailed: [test-control-01] => {"failed": true}[0m
[0;31mmsg: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
[0m
[0;31m
FATAL: all hosts have already failed -- aborting[0m

PLAY RECAP ******************************************************************** 
docker | enable docker ------------------------------------------------- 86.38s
docker | install docker packages --------------------------------------- 46.92s
common | install system utilities -------------------------------------- 21.94s
common | update setuptools and pip ------------------------------------- 19.65s
common | install distributive ------------------------------------------ 15.45s
consul-template | install consul-template ------------------------------- 9.30s
collectd | install collectd packages ------------------------------------ 7.43s
docker | install latest device-mapper-libs ------------------------------ 4.33s
common | enable EPEL repo ----------------------------------------------- 3.77s
common | install pip ---------------------------------------------------- 3.76s
```

After it errors out, if I check the nodes docker is enabled and running.  Then if I re-run the playbook to install mantl it progresses and moves along, most often to completion on the second run.  

Related, I've noticed an impact on how well the setup runs based on the virtual resources the nodes have.  In my case, a 1 CPU, 4 GB RAM setup often errors out multiple times, and in some cases won't work at all.  I'm currently testing with 4CPU/8GB boxes and having no trouble.

**Comment by stevendborrelli** [comment]
@hpreston Thanks for opening this. This might be due to the fact that device mapper takes a while to initialize (longer than systemctl is willing to wait). We are hoping that #922 will speed up this process.

**Comment by siddharthist** [comment]
I have also seen this error intermittently.

**Comment by siddharthist** [comment]
This continues to affect the master branch of Mantl.

**Comment by briantully** [comment]
Still seeing this on the master branch as well.

**Comment by siddharthist** [comment]
I haven't seen this in a while, has anyone else?

**Comment by ryane** [comment]
haven't seen this in a long while

**Comment by Cryptophobia** [comment]
@Theaxiom : I am starting to see this again on the master branch when I run `vagrant up`. Wasn't getting this a couple days ago on the master branch.

**Comment by KaGeN101** [comment]
Must be a race condition then that is triggered again now by changes on master maybe we can do a diff and down what causes it. Does not happen on cloud deploy thou

**Comment by Cryptophobia** [comment]
@KaGeN101 : Hmm, it is something that was added in the past 3 or so days.

**Comment by Theaxiom** [comment]
Probably because laptop is slow

**Comment by Cryptophobia** [comment]
I reverted back around 20 commits (to a8cdf47)  and vagrant deploys works without error:  `git reset --hard a8cdf47` was able to open kubernetes dashboard as well.

I went back to HEAD and it breaks at the same step. So it must be something added in the past 24 hours.

**Comment by KaGeN101** [comment]
Yeah everything works with this work around except the pods cant be on the calico network meaning your mesos containers and k8 containers wont see each other or any pod deployed on the calico network cant get ou

**Comment by KaGeN101** [comment]
@Cryptophobia added a 30 second pause to wait for k8 to settle that is the only timing things added. Very next commit after #96c2d00. May go to #54448e5 and see if it breaks again then we know it is this. Everything after 5444 is not stable currently

**Comment by Cryptophobia** [comment]
@KaGeN101 : #54448e5 fails at the same `enable docker` step. Likely it is the 30 second pause that is causing the issue.

#a8cdf47 works and the k8 dashboard comes up without CrashBackLoop with Vagrant.

However, the latest master deploys on Vagrant but dashboard is in CrashBackLoop.

Should we just tag the code at #a8cdf47 as the vagrant release. :D ?

**Comment by Theaxiom** [comment]
@Cryptophobia What is the error you are receiving? Please paste it here.

**Comment by Theaxiom** [comment]
@Cryptophobia Vagrant works perfectly fine for me at Release 1.3

**Comment by Theaxiom** [comment]
@Cryptophobia could you please get in MM or Gitter so I can assist you? I am pretty sure I know the problem.

**Comment by Cryptophobia** [comment]
@Theaxiom : I am on MM right now. I tried Release 1.3 and it did not work for me. Dashboard crashes with CrashBackLoop at tag 1.3 .

What do you think the issue is? #a8cdf47 works and the k8 dashboard comes up for me on that commit.

**Comment by Cryptophobia** [comment]
@Theaxiom : I get the following error when I get the logs from the Kuberenetes dashboard container on release 1.3.0:

```
Error while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: Get https://kubernetes.service.consul:6443/version: dial tcp 172.20.10.101:6443: i/o timeout
```

**Comment by KaGeN101** [comment]
It is because at #a8cdf47 the workaround was in that puts the container on the k8 host network essentially cutting calico networking out and being on 10.0.0.0/8
There is still an issue if your VM network is not on 10.0.0./8 in calico as is the case with Vagrant

You can add the workaround back placing the containers on the k8 hostNetwork=true if you like and see what that does?

**Comment by thomasvincent** [comment]
@Cryptophobia did  k8 hostNetwork=true solve the issue?

**Comment by Cryptophobia** [comment]
@thomasvincent yes, deploying with the hostNetwork=true fix on a8cdf47 works for me with the latest on master.
